### PR TITLE
[0.10] Rename unsignedPayload and ec2QueryName

### LINKED ISSUE #281: [0.10] Move aws.api#ec2QueryName to aws.protocols#ec2QueryName
Now that 0.10 introduces the `aws.protocols` namespace, it makes more sense for the `ec2QueryName` trait to be defined in that namespace with the rest of the protocol traits rather than `aws.api`.

### PR DESCRIPTION
**Pull request by mtdowling**
Now that the aws.api#protocols and aws.api#auth namespaces exist, these
traits make more sense to be defined in those namespaces.

*Issue #, if available:*
Closes #281 

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Smithy Changelog
 
+## 0.10.0 (2020-??-??)
+
+### Breaking changes
+
+* aws.api#ec2QueryName was renamed to `aws.protocols#ec2QueryName`.
+* aws.api#unsignedPayload was renamed to `aws.auth#unsignedPayload` and no longer takes an array of values.
+  All auth schemes that honor this trait should be consider unsigned when this trait is applied.
```
Comment (kstich) [inline CHANGELOG.md]: s/consider/considered

File: docs/source/spec/aws-query-protocol.rst [hunk 2]
```diff
@@ -102,4 +102,65 @@ Value type
...
+
+.. _aws.protocols#ec2QueryName-trait:
+
+------------------------------------
+``aws.protocols#ec2QueryName`` trait
+------------------------------------
+
+Summary
+    Indicates the serialized name of a structure member when that structure is
+    serialized for the input of an EC2 operation.
```
Comment (kstich) [inline docs/source/spec/aws-query-protocol.rst]: "an operation using the `ec2Query` protocol." ?

File: smithy-aws-traits/README.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ See the Smithy specification for details on how these traits are used.
 * `aws.api#service`: Configures a Smithy service as an AWS service.
 * `aws.api#arn`: Defines the ARN template of a Smithy resource.
 * `aws.api#arnReference`: Indicates that a string shape contains an ARN.
-* `aws.api#unsignedPayload`: Configures that request payload of an
+* `aws.auth#unsignedPayload`: Configures that request payload of an
```
Comment (kstich) [inline smithy-aws-traits/README.md]: This is a list of contents in the `aws.api` namespace, should the README contain new lists for `aws.auth` and `aws.protocols`?

File: smithy-aws-traits/src/test/java/software/amazon/smithy/aws/traits/UnsignedPayloadTraitTest.java [hunk 4]
```diff
@@ -18,6 +18,7 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import org.junit.jupiter.api.Test;
+import software.amazon.smithy.aws.traits.auth.UnsignedPayloadTrait;
```
Comment (kstich) [inline smithy-aws-traits/src/test/java/software/amazon/smithy/aws/traits/UnsignedPayloadTraitTest.java]: Should this instead be in the correct package within `src/test`?

File: smithy-aws-traits/README.md (hunk 3 above)
Comment (mtdowling) [reply smithy-aws-traits/README.md]: I'll just remove the README in favor of the spec.

**Pull request merged**